_P101_TMP117.ino

```diff
@@ -0,0 +1,269 @@
+        optionValues1[3] = (0x03);
+        options1[3] = F("3 - Averaging (64)");
+        
+        string += F("<TR><TD>Averaging:<TD><select name='plugin_101_Averaging'>");
+        for (byte x = 0; x < TMP117_Averaging_OPTION; x++)
+        {
+          string += F("<option value='");
+          string += optionValues1[x];
+          string += "'";
+          if (choice1 == optionValues1[x])
+            string += F(" selected");
+          string += ">";
+          string += options1[x];
+          string += F("</option>");
+        }
+        string += F("</select>");
+        string += F("<TR><TD>Sensor Adjustment [&#176;C]:<TD><input type='text' title='calibrate sensor to target temperature' name='");
+        string += F("plugin_101_Adjust' value='");
+        string += Settings.TaskDevicePluginConfigFloat[event->TaskIndex][2];
+        string += F("'>");
+        string += F("<TR><TD>Offset [&#176;C]:<TD><input type='text' title='set temperature offset like ' name='");
+        string += F("plugin_101_Offset' value='");
+        string += Settings.TaskDevicePluginConfigFloat[event->TaskIndex][3];
+        string += F("'>");
+        
+       // LoadTaskSettings(event->TaskIndex); // we need to restore our original taskvalues!
+        success = true;
+        break;
+      }
+
+    case PLUGIN_WEBFORM_SAVE:
+      {
+        String plugin1 = WebServer.arg("plugin_101_ConvTime");
+        Settings.TaskDevicePluginConfig[event->TaskIndex][0] = plugin1.toInt();
+        String plugin2 = WebServer.arg("plugin_101_Averaging");
+        Settings.TaskDevicePluginConfig[event->TaskIndex][1] = plugin2.toInt();
+        String adjust = WebServer.arg("plugin_101_Adjust");
+        Settings.TaskDevicePluginConfigFloat[event->TaskIndex][2] = adjust.toFloat();
+        String offset = WebServer.arg("plugin_101_Offset");
+        Settings.TaskDevicePluginConfigFloat[event->TaskIndex][3] = offset.toFloat();
+        
+        Plugin_101_init = false; // Force device setup next time
+        success = true;
+        break;
+      }
+         
+   case PLUGIN_INIT:
+    {
+      Plugin_101_init = true;
+      success = true;
+      break;
+  }
+  
+    case PLUGIN_READ:
+      {
+ /*  command   parameter   description
+   *    0           X           print actual temperature
+   *    1           X           print EEPROM NIST UUID [E1|E2|E3]
+   *    2         float         set temperature offset like "2 20.5" sets the offset to 20.5°C
+   *    3         float         calibrate sensor to target temperature like "3 30.5" will calibrate sensor to 30.5°C
+   * 
+   */
+        // Get sensor resolution configuration
+         uint8_t ConvTime = Settings.TaskDevicePluginConfig[event->TaskIndex][0];
+         uint8_t Averaging = Settings.TaskDevicePluginConfig[event->TaskIndex][1];
+         float adjust = Settings.TaskDevicePluginConfigFloat[event->TaskIndex][2];
+         float offset = Settings.TaskDevicePluginConfigFloat[event->TaskIndex][3];

```

Comment:
You're declaring variables here which have the same name and signature as the global variables at the beginning of this file. 
If they are only needed here, then remove the global declared ones.